js/blockTrackers.js

```diff
@@ -11,13 +11,28 @@ var entityList = JSON.parse(load.loadExtensionFile(entityListSource, 'json', 'ex
 
 var betterList = JSON.parse(load.loadExtensionFile('better-pages.txt', 'json'));
 
+var bg = chrome.extension.getBackgroundPage();
+var formerSocialBlocking = bg.isSocialBlockingEnabled;
+
 require.scopes.blockTrackers = (function() {    
     
     // If blocking option is enabled
     // and url matches a tracker pattern
     // block the request
     function blockTrackers(url, currLocation) {
         if (localStorage['blocking'] === 'true') {
+            if (formerSocialBlocking !== bg.isSocialBlockingEnabled) {
+                formerSocialBlocking = bg.isSocialBlockingEnabled;
+                trackers = load.processMozillaBlockList(blockList);
```

Comment:
Got it. Maybe it's worth restructuring the `trackers` dictionary to include categories as a top level? That would make it easier to turn it off/on with out rebuilding the list.